### PR TITLE
Add new cop `Minitest/RefuteNil`

### PR DESCRIPTION
**Pull request by tejasbubane**
* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](https://github.com/rubocop-hq/rubocop-minitest/blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop-hq/rubocop-minitest/blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [x] Run `bundle exec rake default`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Comment by tejasbubane** [comment]
I created a separate cop for now but should we have separate refute cops or have single cop which handles refute as well? This will apply to almost all other cops.

**Comment by koic** [comment]
Ah, that's a good point of view. RuboCop Minitest can be designed to provide a single cop that bundles `refute_*`.

However, a design that provides individual cops along The Minitest Style Guide may be fine. So, Minitest is a simple API and expects a limited number of style rules.

I think that it may be simple if a rule of Minitest Style Guide and a cop is mapped. Each cop can have a reference to a rule of The Minitest Style Guide.

Cc @rubocop-hq/rubocop-core

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 ## master (unreleased)
 
+* [#11](https://github.com/rubocop-hq/rubocop-minitest/pull/11): Add new `Minitest/RefuteNil` cop. ([@tejasbubane ][])
```
Comment (koic) [inline CHANGELOG.md]: Can you add `### New features` category?

**Comment by koic** [comment]
Please add this cop setting to config/default.yml.

File: CHANGELOG.md (hunk 1 above)
Comment (tejasbubane) [reply CHANGELOG.md]: added

**Comment by koic** [comment]
Thanks!

File: config/default.yml [hunk 2]
```diff
@@ -7,3 +7,9 @@ Minitest/AssertNil:
   StyleGuide: 'https://github.com/rubocop-hq/minitest-style-guide#assert-nil'
   Enabled: true
   VersionAdded: '0.1'
+
+Minitest/RefuteNil:
+  Description: 'Check if your test uses `refute_nil` instead of `refute_equal(nil, something)`.'
+  StyleGuide: 'https://github.com/rubocop-hq/minitest-style-guide#refute-nil'
+  Enabled: true
+  VersionAdded: '0.1.1'
```
Comment (koic) [inline config/default.yml]: I overlooked the version and updated it by https://github.com/rubocop-hq/rubocop-minitest/commit/b9c8da6. FYI, Minor version will be bumped when new cop are added.

**Pull request merged**